### ISSUE TITLE
Maxed out "Smithing" skill makes the Anvil unusable

### ISSUE THREAD

**Issue by tszkoda** [Globox1997/LevelZ#196]
I noticed an issue with the "Smithing" skill in LevelZ:

If you max out your smithing skill (level 40, by default), there is an issue that occurs when trying to use an anvil.

Please see the below screenshots as an example:

![LevelZ bug 1](https://user-images.githubusercontent.com/55335305/196275170-9f0d8c69-2cb1-465f-99f0-336cf55981f9.PNG)

![LevelZ bug 2](https://user-images.githubusercontent.com/55335305/196275344-98f85050-57ab-4048-8d4e-35346f0be841.PNG)

As you can see in the above screenshot, if the anvil xp cost is low enough, it allows you to use the anvil as usual.

![LevelZ bug 3](https://user-images.githubusercontent.com/55335305/196275595-49df9846-3dd9-4deb-99ec-904964dc1e8a.PNG)

However, if the xp cost of the repair gets larger (like in the above screenshot), the xp cost goes to 0 and it will **NOT** allow you to repair the item.

This same issue applies to any anvil use case, such as trying to add an enchanted book to a tool.

This issue is strange because it seems like the anvil should allow you to repair the item at no xp cost (like in the last screenshot), but when you try to take it out of the anvil, it won't let you.

Thank you in advance for addressing this bug.

*Please note that these tests were taken while using the LevelZ mod within the Valhelsia: Enhanced Vanilla modpack, version 1.4.2b

**Comment by Globox1997** [comment]
the second screenshot looks like it just cost nothing but you can't pull it out?

**Comment by Globox1997** [comment]
try this pls
[levelz-1.4.1.zip](https://github.com/Globox1997/LevelZ/files/9809298/levelz-1.4.1.zip)
you have to unzip it

**Comment by tszkoda** [comment]
> the second screenshot looks like it just cost nothing but you can't pull it out?

Yes, in the last screenshot, I can't pull out the repaired Diamond Chestplate. I'm guessing that the xp cost is so low that it breaks the use of the anvil for some reason.




> try this pls [levelz-1.4.1.zip](https://github.com/Globox1997/LevelZ/files/9809298/levelz-1.4.1.zip) you have to unzip it

So, I tried to use that version of levelz with the Valhelsia modpack, but it crashed on startup because of an incompatibility issue. This version of Valhelsia: Enhanced Vanilla (1.4.2b) runs on Minecraft 1.18.2, but that LevelZ file (1.4.1) looks like it runs only on the new 1.19.x Minecraft updates. 

It looks like this modpack was originally using the 1.3.3 version of LevelZ

Here is the crash report:
![LevelZ compatibility error](https://user-images.githubusercontent.com/55335305/196510036-d9a616fa-7b05-4bb5-aa66-e400d7fd50bd.PNG)

**Comment by tszkoda** [comment]
> try this pls [levelz-1.4.1.zip](https://github.com/Globox1997/LevelZ/files/9809298/levelz-1.4.1.zip) you have to unzip it

I wanted to let you know that I did a seperate test, where I ran the 1.4.1 version of LevelZ in Fabric Minecraft 1.19.2 and **it worked**.

I made my smithing skill level 40 and I tried to repair an iron helmet. I used 2 iron ingots and the xp cost was 0 (just like in the previous diamond armor screenshot), but this time the anvil allowed me to repair the item and put it back into my inventory.

This outcome makes me think that there must be a bug in the 1.3.3 version of LevelZ.

I would have used the 1.4.1 version of LevelZ in the Valhelsia modpack, but since the Minecraft version is different, it doesn't want to work.

Please let me know if there is anything else I can do to help. Thanks again!
